packages/nameguard-react/src/components/ReportBadge/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { type ConsolidatedNameGuardReport } from "@namehash/nameguard";
+import React from "react";
+import cc from "classcat";
+
+import { ReportIcon } from "../ReportIcon/index";
+import { RatingLoadingIcon, RatingIconSize } from "../..";
+import { UnknownReportShield } from "../UnknownReportShield/UnknownReportShield";
+
+interface ReportBadgeProps {
+  name: string;
```

Comment:
This UI component should always validate that if `data` is not null that `data.name === name`. If this isn't true then it should throw an Error.

This idea should also be documented in /** ... **/ comments above the UI component.